sqlglot/optimizer/simplify.py

```diff
@@ -369,6 +370,51 @@ def absorb_and_eliminate(expression, root=True):
     return expression
 
 
+def propagate_constants(expression, root=True):
+    """
+    Propagate constants for conjunctions in DNF:
+
+    SELECT * FROM t WHERE a = b AND b = 5 becomes
+    SELECT * FROM t WHERE a = 5 AND b = 5
+
+    Reference: https://www.sqlite.org/optoverview.html
+    """
+    from sqlglot.optimizer.normalize import normalized
```

Comment:
Kind of a smell that we need a circular dependency here.
Maybe `normalized` should be in its own module, separate from `normalize`.

Reply:
Yeah.. I tried to move `normalized` into `helper` at first but then I needed to do a local import because of another circular dependency with `from sqlglot import exp` so I just left it as shown here..

Reply:
Moved `normalized` in `expressions.py` since it can also be viewed as a general-purpose "expression helper".

Reply:
Ended up moving it back to `normalize.py` because of the bug I mentioned below - we needed `find_all_in_scope` which was also causing circular dependency issues... 🤦 

I ended up doing `import sqlglot` to refer to `normalized` using the full module path .